java/flight/src/main/protobuf/flight.proto

```diff
@@ -0,0 +1,260 @@
+  
+  /*
+   * For a given FlightDescriptor, get information about how the flight can be consumed. This is a
+   * useful interface if the consumer of the interface already can identify the specific flight to
+   * consume. This interface can also allow a consumer to generate a flight stream through a specified
+   * descriptor. For example, a flight descriptor might be something that includes a SQL statement or
+   * a Pickled Python operation that will be executed. In those cases, the descriptor will not be
+   * previously available within the list of available streams provided by ListFlights but will be
+   * available for consumption for the duration defined by the specific flight service.
+   */  
+  rpc GetFlightInfo(FlightDescriptor) returns (FlightGetInfo) {}
+  
+  /*
+   * Retrieve a single stream associated with a particular descriptor associated with the referenced
+   * ticket. A Flight can be composed of one or more streams where each stream can be retrieved using
+   * a separate opaque ticket that the flight service uses for managing a collection of streams.
+   */
+  rpc DoGet(Ticket) returns (stream FlightData) {}
+
+  /*
+   * Push a stream to the flight service using associated with a particular flight stream. This allows
+   * a client of a flight service to upload a stream of data. Depending on the particular flight
+   * service, a client consumer could be allowed to upload a single stream per descriptor or an
+   * unlimited number. (In the latter, the service might implement a 'seal' action that can be applied
+   * to a descriptor once all streams are uploaded.
+   */  
+  rpc DoPut(stream FlightData) returns (PutResult) {}
+  
+  /*
+   * Flight services can support an arbitrary number of simple actions in addition to the possible
+   * ListFlights, GetFlightInfo, DoGet, DoPut operations that are potentially available. DoAction
+   * allows a flight client to do a specific action against a flight service. An action includes
+   * opaque request and response objects that are specific to the type action being undertaken.
+   */  
+  rpc DoAction(Action) returns (stream Result) {}
+  
+  /*
+   * A flight service exposes all of the available action types that it has along with descriptions.
+   * This allows different flight consumers to understand the capabilities of the flight servic
+   */
+  rpc ListActions(Empty) returns (stream ActionType) {}
+  
+}
+
+/*
+ * The request that a client provides to a server on handshake.
+ */
+message HandshakeRequest {
+
+  /*
+   * A defined protocol version
+   */
+  uint64 protocol_version = 1;
+  Credential credential = 2;
+  bytes payload = 3;
+}
+
+message HandshakeResponse {
+  uint64 protocol_version = 1;
+  bytes payload = 2;
+}
+
+message Credential {
+  enum CredentialType {
+    UNKNOWN = 0;
+    BASIC = 1;
+    OPAQUE = 2;
+  }
+  
+  CredentialType type = 1;
+  string username = 2;
+  string password = 3;
+  bytes opaque_value = 4;
```

Comment:
Maybe add a comment the username/password is mutually exclusive from opaque_value

Reply:
I changed this so the handshake only has an opaque value.